netbox/ipam/forms/model_forms.py

```diff
@@ -362,15 +362,25 @@ def clean(self):
 
         # Do not allow assigning a network ID or broadcast address to an interface.
         if interface and (address := self.cleaned_data.get('address')):
-            if address.ip == address.network:
-                msg = f"{address} is a network ID, which may not be assigned to an interface."
-                if address.version == 4 and address.prefixlen not in (31, 32):
+            prefix_str = f"{self.instance.address.network}/{self.instance.address.prefixlen}"
+            allow_assignment_error = True
+            if self.instance.vrf is None:
+                prefix_obj = Prefix.objects.get(prefix=prefix_str)
```

Comment:
Wont this throw an exception if no prefix exists?

Reply:
I will double check tomorrow am but i thought i tested that.